core/src/main/scala/org/apache/spark/deploy/security/ServiceCredentialProvider.scala

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.spark.deploy.yarn.security
+package org.apache.spark.deploy.security
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
```

Comment:
Hmm... that's unfortunate, not just because of this change, but because `yarn/` is not included in mima checks, so if the class changed in incompatible ways we wouldn't notice.

I really doubt anyone has custom implementations of that interface at this point. And making it a proper API in core is better going forward, IMO. If you really think that it's important to keep backwards compatibility, we could have the old interface present, extending the new one, and some compatibility code to keep both working. But that sounds like a lot of work for something I doubt anyone is currently using...